modules/extensions/galasa-extensions-parent/dev.galasa.ras.couchdb/src/main/java/dev/galasa/ras/couchdb/internal/CouchdbDirectoryService.java

```diff
@@ -432,76 +469,58 @@ private JsonArray buildQuerySortJson(@NotNull RasSortField primarySort) {
     }
 
-    private void discardRecord(String databaseName, String id) throws ResultArchiveStoreException {
-        discardRecord(databaseName, id, getRevision(databaseName, id));
+    private CompletableFuture<Void> discardRecord(String databaseName, String id) throws ResultArchiveStoreException {
+        return getRevision(databaseName, id)
+            .thenCompose(revision -> discardRecord(databaseName, id, revision));
     }
 
-    private void discardRecord(String databaseName, String id, String revision) throws ResultArchiveStoreException {
-        URIBuilder builder;
-        try {
-            builder = new URIBuilder(store.getCouchdbUri() + "/" + databaseName + "/" + id);
-            builder.addParameter("rev", revision);
-
-            HttpDelete httpDelete = requestFactory.getHttpDeleteRequest(builder.build().toString());
-
-            try (CloseableHttpResponse response = store.getHttpClient().execute(httpDelete)) {
-                StatusLine statusLine = response.getStatusLine();
-
-                if (statusLine.getStatusCode() != HttpStatus.SC_OK) {
-                    throw new CouchdbRasException("Unable to delete run "+ id +" - " + statusLine.toString());
-                }
-            } catch (IOException e) {
-                throw new ResultArchiveStoreException("Failed to execute HTTP DELETE for run: " + id, e);
+    private CompletableFuture<Void> discardRecord(String databaseName, String id, String revision) {
+        return CompletableFuture.runAsync(() -> {
+            try {
+                store.deleteDocumentFromDatabase(databaseName, id, revision);
+            } catch (CouchdbException e) {
+                throw new CompletionException(e);
             }
-        } catch (URISyntaxException e) {
-            throw new ResultArchiveStoreException(e);
-        }
+        });
     }
 
-    private String getRevision(String databaseName, String id) throws ResultArchiveStoreException {
-        HttpGet httpGet = requestFactory.getHttpGetRequest(store.getCouchdbUri() + "/"+ databaseName+"/"+id);
-
-        try (CloseableHttpResponse response = store.getHttpClient().execute(httpGet)) {
-            StatusLine statusLine = response.getStatusLine();
-            HttpEntity entity = response.getEntity();
-            String responseEntity = EntityUtils.toString(entity);
-
-            if (statusLine.getStatusCode() != HttpStatus.SC_OK) {
-                throw new CouchdbRasException("Unable to find runs - " + statusLine.toString());
-            }
-
-            IdRev found = store.getGson().fromJson(responseEntity, IdRev.class);
-            if (found._id == null) {
-                throw new CouchdbRasException("Unable to find runs - Invalid JSON response");
-            }
-            if (found._rev == null) {
-                throw new CouchdbRasException("Unable to find rev - Invalid JSON response");
+    private CompletableFuture<String> getRevision(String databaseName, String id) {
```

Comment:
I might have been tempted to separate the async/future stuff into a separate class. This class is getting huge. Or even just separate out the deletion implementation as it's kinda separate/doing things in a separate way to the rest ?

Reply:
Agreed - moved the deletion code into a separate class in https://github.com/galasa-dev/galasa/pull/108/commits/fa9d1a5443776870efa5b5cf710205d2e2999fcb